espresso/core/java/androidx/test/espresso/util/SpecialCharVisualizer.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.test.espresso.util;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import org.hamcrest.Description;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * An auxiliary class that makes masked characters more visible.
+ * You might want to use it in cases when user thorough comprehension is required, i.e. test reports.
+ */
+public class SpecialCharVisualizer {
+
+  private static final Map<Character, String> SPECIAL_CHAR_CODES = new HashMap<>();
+  static {
+    SPECIAL_CHAR_CODES.put('\u00A0', "[NBSP]");
+  }
+
+  @NonNull
+  public static String highlight(@NonNull CharSequence text) {
+    StringBuilder result = new StringBuilder();
+    for (int i = 0; i < text.length(); i++) {
+      Character current = text.charAt(i);
+      result.append(SPECIAL_CHAR_CODES.containsKey(current) ? SPECIAL_CHAR_CODES.get(current) : current);
+    }
+    return result.toString();
+  }
+
+  public static Description appendHighlightedValue(
+      @Nullable CharSequence text, Description description, Boolean asValue
+  ) {
+    if (text == null) {
+      return description;
+    }
+    String result = SpecialCharVisualizer.highlight(text);
+    if (asValue) {
+      description.appendValue(result);
+    } else {
+      description.appendText(result);
+    }
+    if (!result.contentEquals(text)) {
+      description.appendText(
+          " (hint: this string contains special characters, see androidx.test.espresso.util.SpecialCharVisualizer.java)"
```

Comment:
Thoughts on a shorter version?
```suggestion
          " (This string contains special characters - see SpecialCharVisualizer in Espresso.)"
```
(class might be konverted later, and no need for FQCN, Google/jump to class/file will find it anyway)

Reply:
I preferred FQCN with the older class name, as `SpecialCharacter` seemed to be too commonplace.

`hint` prefix comes across as some flag for visual difference right after the text, I thought it vital

However, I don't see new version loses any readability points, it's ok to fix.